api/app/controllers/spree/api/promotions_controller.rb

```diff
@@ -3,26 +3,18 @@
 module Spree
   module Api
     class PromotionsController < Spree::Api::BaseController
-      before_action :requires_admin
       before_action :load_promotion
 
       def show
-        if @promotion
-          respond_with(@promotion, default_template: :show)
-        else
-          raise ActiveRecord::RecordNotFound
-        end
+        authorize! :read, @promotion
+        respond_with(@promotion, default_template: :show)
       end
 
       private
 
-      def requires_admin
-        return if @current_user_roles.include?("admin")
-        unauthorized && return
-      end
-
       def load_promotion
-        @promotion = Spree::Promotion.find_by(id: params[:id]) || Spree::Promotion.with_coupon_code(params[:id])
+        @promotion = Spree::Promotion.with_coupon_code(params[:id])
+        @promotion ||= Spree::Promotion.find(params[:id])
```

Comment:
@SamuelMartini for consistency with what was done after [this](https://github.com/solidusio/solidus/pull/3529#discussion_r386952504) I would consider using the `||` here as well, WDYT?

Reply:
@spaghetticode changed in https://github.com/solidusio/solidus/pull/3529/files#diff-208838646d38e5237d91319be1b4d136R16